pkg/logql/ast.go

```diff
@@ -148,22 +149,28 @@ func (m MultiStageExpr) String() string {
 
 func (MultiStageExpr) logQLExpr() {} // nolint:unused
 
-type matchersExpr struct {
+type MatchersExpr struct {
 	matchers []*labels.Matcher
 	implicit
 }
 
-func newMatcherExpr(matchers []*labels.Matcher) *matchersExpr {
-	return &matchersExpr{matchers: matchers}
+func newMatcherExpr(matchers []*labels.Matcher) *MatchersExpr {
+	return &MatchersExpr{matchers: matchers}
 }
 
-func (e *matchersExpr) Matchers() []*labels.Matcher {
+func (e *MatchersExpr) Matchers() []*labels.Matcher {
 	return e.matchers
 }
 
-func (e *matchersExpr) Shardable() bool { return true }
+func (e *MatchersExpr) AppendMatchers(m []*labels.Matcher) {
+	e.matchers = append(e.matchers, m...)
```

Comment:
I'm wondering if we should get the matchers just public. WDYT ?

Reply:
Yeah I like this suggestion more as well.

Reply:
AFAICS, none of the expression structs exposes any of their field public, e.g. some interesting field would be `LiteralExpr.value`, `*AggrevationExpr.params`, `OffsetExpr.offset`. IMHO ASTs are very sensitive to public scoping of internals. ASTs are often subject of optimizations and thus keeping fields private is a good choice to save your options for later. WDYT?

Reply:
Both sound fine to me 👍

Reply:
I think ultimately we'll make those public if we move to another package.

Reply:
Let's move them to public when we and if we polish this to another package.